xbmc/network/Socket.cpp

```diff
@@ -171,6 +138,42 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
   return true;
 }
 
+bool CPosixUDPSocket::CheckIPv6(int port, int range)
+{
+  bool ipv6 = false;
+  CAddress testaddr("::");
+
+  SOCKET testSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+  if (testSocket == INVALID_SOCKET)
+    return false;
+
+#ifdef WINSOCK_VERSION
+  const char zero = 0;
+#else
+  int zero = 0;
+#endif
+
+  if (setsockopt(testSocket, IPPROTO_IPV6, IPV6_V6ONLY, &zero, sizeof(zero)) == -1)
+  {
+    closesocket(testSocket);
+    return false;
+  }
+
+  // Try to bind a socket to validate ipv6 status
+  for (; port <= port + range; port++)
+  {
+    testaddr.saddr.saddr6.sin6_port = htons(port);
+    if (bind(testSocket, reinterpret_cast<struct sockaddr*>(&testaddr.saddr), testaddr.size) >= 0)
+    {
+      ipv6 = true;
```

Comment:
I see the ipv6 bool is just there, so that one can do another closesocket before returning. Scope Guard would be nice, but it's okay as is.

Reply:
I put a commit on top that's using ScopeGuard, if we want to keep it I'll squash.
Otherwise I'll drop it, it's a but unfortunate that you can't feed it with the closesocket define :(

Reply:
No, no. Keep it as is. The other commit can come on top, afterwards :-)

Reply:
argh, there is still an error in it..